### PR TITLE
llvm/clang/lldb-9.0: new port

### PR DESCRIPTION
**Pull request by kencu**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.6 18G95
Xcode 10.3 10G8 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by kencu** [comment]
This document appears to no longer exist; at least it is not in that place any longer, and I can't seem to find it anywhere.
```
${worksrcpath}/tools/lldb/docs/code-signing.txt
```
We can crib one out of lldb-8.0 and put it in the files directory if you like.

Otherwise, it installs, runs on 10.14. Will obviously need further testing, but I thought this might get the ball rolling and relieve @jeremyhu of at least a little bit of work.

@cjones051073 , @MarcusCalhoun-Lopez

**Comment by kencu** [comment]
I have not as yet run the builds back on any older systems.

**Comment by kencu** [comment]
Looks like the 7.3 buildbot for travis is "Back to the Future":
```
DEBUG: Portfile is from the future - check date and time of your system
    while executing
"open_statefile"
    (procedure "check_variants" line 28)
    invoked from within
"check_variants fetch"
    invoked from within
"$workername eval "check_variants $target""
    (procedure "mportexec" line 7)
    invoked from within
"mportexec $workername $target"
Error: Unable to execute port: Portfile is from the future - check date and time of your system
Fetch of dependency 'help2man' with variants '' failed, aborting.
```

**Comment by cjones051073** [comment]
Thanks @kencu, very nice to have this.

Where do you stand with 10.15 b.t.w. ? Have you accepted the NDA for it ? Been trying clang on this beta OS and there are a few issues, but I would rather discuss them privately if you are willing to ?

**Comment by kencu** [comment]
The code signing document was removed here <https://github.com/llvm/llvm-project/commit/a4210f7b102984de1b38f0f45abf3e9f1be0321d#diff-efa325aee25f0bc2c5b6b25bd4c14bcf> so I guess we can just delete that part of the Portfile, possibly with the note changed to direct users to <https://lldb.llvm.org/resources/build.html#code-signing-on-macos>

**Comment by kencu** [comment]
> Thanks @kencu, very nice to have this.
> 
> Where do you stand with 10.15 b.t.w. ? Have you accepted the NDA for it ? Been trying clang on this beta OS and there are a few issues, but I would rather discuss them privately if you are willing to ?

I have not as yet started into 10.15; trying to see how I would like to approach it at the moment. I might set it up in a VM I suppose; I'm not sure I want to limit my primary behemoth to no 32bit software...

**Comment by kencu** [comment]
that latest push fixes the clang-9.0 build on 10.6.8

File: lang/llvm-9.0/Portfile [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+patchfiles \
+    0001-Set-the-Mach-O-CPU-Subtype-to-ppc7400-when-targeting.patch \
+    0002-Define-EXC_MASK_CRASH-and-MACH_EXCEPTION_CODES-if-th.patch \
+    0003-MacPorts-Only-Don-t-embed-the-deployment-target-in-t.patch \
+    0004-Fix-build-issues-pre-Lion-due-to-missing-a-strnlen-d.patch \
+    0005-Threading-Only-call-pthread_setname_np-on-SnowLeopar.patch \
+    0006-Only-call-setpriority-PRIO_DARWIN_THREAD-0-PRIO_DARW.patch
+
+# added by kencu for older systems support
+patchfiles-append \
```
Comment (kencu) [inline lang/llvm-9.0/Portfile]: Jeremy, these three patches can be rolled into your rebasing repo if you like. I named them 5000-series so you would not confuse them with any of your own. There is one more down below that is a clang-only patch, that I wrapped in conditional.

Comment (kencu) [inline lang/llvm-9.0/Portfile]: Jeremy, these 5000-series patches are for older systems support. The could be rolled into your rebasing repo if you prefer (better).

File: lang/llvm-9.0/Portfile [hunk 2]
```diff
@@ -0,0 +1,715 @@
...
+        patchfiles-append leopard-no-asan.patch
+        configure.args-append -DCOMPILER_RT_BUILD_SANITIZERS=OFF
+    }
+
+    if {${subport} eq "clang-${llvm_version}" && ${os.major} <= 9} {
+        patchfiles-append leopard-no-blocks.patch
+    }
+
+    if {${subport} eq "clang-${llvm_version}" && ${os.major} <= 10} {
+        # the directorywatcher code in clang-9.0+ is 10.7+
```
Comment (kencu) [inline lang/llvm-9.0/Portfile]: one more 5000-series patch I added, clang only.

File: lang/llvm-9.0/Portfile [hunk 3]
```diff
@@ -0,0 +1,715 @@
...
+revision                0
+subport                 clang-${llvm_version} { revision 0 }
+subport                 lldb-${llvm_version} { revision 0 }
+set suffix              mp-${llvm_version}
+set sub_prefix          ${prefix}/libexec/llvm-${llvm_version}
+dist_subdir             llvm
+categories              lang
+platforms               darwin
+license                 NCSA
+maintainers             {jeremyhu @jeremyhu} {larryv @larryv}
```
Comment (cjones051073) [inline lang/llvm-9.0/Portfile]: @kencu I would propose you add yourself as a maintainer as well here ?

**Comment by cjones051073** [comment]
@kencu I have now built this myself on a few OS versions, and built a few reasonably large c++ ports with it, with no problems. So it seems to me to at least in the face if it be working fine.

So how do you want proceed with merging ? I guess ideally you would get @jeremyhu ’s blessing first, but we know it could be a long time before he replies here, and I think it would be nice to get this released, personally.

**Comment by kencu** [comment]
I know -- we always lose Jeremy around new Xcode releases, and one just came out as you know. Let's give him a couple of days, and then we'll merge if he has no time. I think it's OK too, but Jeremy always knows plenty more than I do about things, so I like to stay well on the side of super-careful with these compilers.

**Comment by cjones051073** [comment]
> I know -- we always lose Jeremy around new Xcode releases, and one just came out as you know. Let's give him a couple of days, and then we'll merge if he has no time. I think it's OK too, but Jeremy always knows plenty more than I do about things, so I like to stay well on the side of super-careful with these compilers.

Totally understood. It is though I think worth bearing in mind this update has limited scope for causing problems, as nothing in MP will use it to build ports, for instance, at least at first. So in that sense it’s arguably way safer to merge this, than my recent update to 8.0.1 which if wrong would have caused a number of headaches.... If Jeremy does eventually object to somehting it can always be changed in an update.

**Comment by pmetzger** [comment]
>  It is though I think worth bearing in mind this update has limited scope for causing problems, as nothing in MP will use it to build ports, for instance, at least at first. So in that sense it’s arguably way safer to merge this, than my recent update to 8.0.1 which if wrong would have caused a number of headaches.... If Jeremy does eventually object to something it can always be changed in an update.

I think this is reasonable. If Jeremy eventually surfaces, he can correct what we did then, and it will cause no harm. Also, this port isn't yet used for anything so it does no harm to commit. It's been almost two years since Jeremy was giving us significant feedback, FWIW.

**Comment by kencu** [comment]
It seems like that -- he's heavy at Apple, and has a new family -- but perhaps you didn't notice that he was around just a few weeks ago, when I asked him about the ramifications of changing the id's on all the clang/llvm/lldb libs to @rpaths to fix our analyzer problems. I'm glad to see that you're all comfortable with how I'm moving things along -- that makes me feel reassured about what I'm doing here. Perhaps there will come a day when I take over this role...I find compilers uber-cool...

**Comment by cjones051073** [comment]
Can I propose you give it another few days for comments, and then when a week has elapsed since you open this you just go ahead and merge. I don’t really see what harm can be done, that could not be fixed up later on... Personally I would like to get this in, as I would like to start using it in a few places....

**Comment by kencu** [comment]
> Can I propose you give it another few days for comments, and then when a week has elapsed since you open this you just go ahead and merge. I don’t really see what harm can be done, that could not be fixed up later on... Personally I would like to get this in, as I would like to start using it in a few places....
@jeremyhu just reviewed my PR in macports-base 10 minutes ago....

**Comment by cjones051073** [comment]
@kencu believe it or not I was writing that comment just as you reassigned to @jeremyhu , so i did not see it before posting.... If he indeed is free to comment soon thats great.

**Comment by pmetzger** [comment]
I certainly would love to see his comments.

**Comment by jeremyhu** [comment]
Thanks for taking initiative here, Ken.  I haven't had a chance to look through this yet, but will over the next week.  I really wish I had more time to help out here, but somehow having a toddler and an infant has a way of sucking up all my free time ;0

**Comment by cjones051073** [comment]
@jeremyhu just wondering if you got a chance yet to look over this ? Would be good to get it merged soon, if possible.

**Comment by kencu** [comment]
let's go ahead with this as-is, then. It seems solid to those of us who have used it, and it's not the fault on any systems. If some kind of error shows up, we can deal with it.

**Comment by cjones051073** [comment]
@kencu FYI build fails in the buildbots on 10.6 (maybe others too).

https://build.macports.org/builders/ports-10.6_x86_64-builder/builds/2214/steps/install-port/logs/stdio

Perhaps fixable by using macports provided python ?

**Comment by kencu** [comment]
Thanks for noticing -- absolutely right. I had to do that in webkit2-gtk as well. I must have already had a newer `python27` installed.

**Pull request merged**